### PR TITLE
Fix Wallet send of ETH and tokens

### LINKED ISSUE #4650: Gas estimation in wallet is not working
### Description

When sending transaction from wallet, the gas is not estimated

image

### PR DESCRIPTION
**Pull request by jrainville**
Fixes #4650 and more
 
This started as a fix for the Gas estimation, but I found that the whole wallet Send was not hooked, so I did it all before @anastasiyaig could go ahead and complain (jk... unless 🤫 )

There will still be remaining work, but the normal flow to send ETH and tokens (tested with STT) works.

The remaining work includes:
- Making the service functions async and hook the Connections again
- Adding a debounce to the calls for estimatingGas and getting the fiatPrice would be good too, because it causes a small lag when typing a value currently

### CODE REVIEW

**Comment by alaibe** [comment]
@Khushboo-dev-cpp you might be interested to look at this PR for your task

File: src/app/modules/main/wallet_section/transactions/controller.nim [hunk 1]
```diff
@@ -78,3 +78,15 @@ method loadTransactions*(self: Controller, address: string, toBlock: Uint256, li
 
 method estimateGas*(self: Controller, from_addr: string, to: string, assetAddress: string, value: string, data: string): string =
   result = self.transactionService.estimateGas(from_addr, to, assetAddress, value, data)
+
+method transferEth*(self: Controller, from_addr: string, to_addr: string, value: string,
+  gas: string, gasPrice: string, maxPriorityFeePerGas: string, maxFeePerGas: string,
+  password: string, uuid: string): bool =
```
Comment (alaibe) [inline src/app/modules/main/wallet_section/transactions/controller.nim:84]: nit: make better indentation
```
method transferEth*(self: Controller, from_addr: string, to_addr: string, value: string,
  gas: string, gasPrice: string, maxPriorityFeePerGas: string, maxFeePerGas: string,
  password: string, uuid: string
): bool =
```

File: src/app/modules/main/wallet_section/transactions/controller_interface.nim [hunk 2]
```diff
@@ -32,6 +32,17 @@ method setTrxHistoryResult*(self: AccessInterface, historyJSON: string) {.base.}
 method estimateGas*(self: AccessInterface, from_addr: string, to: string, assetAddress: string, value: string, data: string): string {.base.} =
   raise newException(ValueError, "No implementation available")
 
+method transferEth*(self: AccessInterface, from_addr: string, to_addr: string, value: string,
+    gas: string, gasPrice: string, maxPriorityFeePerGas: string, maxFeePerGas: string,
+    password: string, uuid: string): bool {.base.} =
```
Comment (alaibe) [inline src/app/modules/main/wallet_section/transactions/controller_interface.nim:37]: nit, indent is 2 spaces, not 4

File: src/app/modules/main/wallet_section/transactions/io_interface.nim [hunk 3]
```diff
@@ -43,6 +43,17 @@ method setIsNonArchivalNode*(self: AccessInterface, isNonArchivalNode: bool) {.b
 method estimateGas*(self: AccessInterface, from_addr: string, to: string, assetAddress: string, value: string, data: string): string {.base.} =
   raise newException(ValueError, "No implementation available")
 
+method transferEth*(self: AccessInterface, from_addr: string, to_addr: string, value: string,
+    gas: string, gasPrice: string, maxPriorityFeePerGas: string, maxFeePerGas: string,
+    password: string, uuid: string): bool {.base.} =
```
Comment (alaibe) [inline src/app/modules/main/wallet_section/transactions/io_interface.nim:48]: same as before, I will stop there on the indentation ;) but the most of the function suffers from it

**Review by alaibe — APPROVED** [review]
LGTM

File: src/app/modules/main/wallet_section/transactions/io_interface.nim (hunk 3 above)
Comment (alaibe) [reply src/app/modules/main/wallet_section/transactions/io_interface.nim:48]: I realize there is no real code style in Nim, but from python (which uses same syntax) this is what most of the autoformatter does (there is 4)

**Comment by anastasiyaig** [comment]
@abhi-status i have reassigned this task to myself because it belongs to wallet cohort which i am in

**Comment by anastasiyaig** [comment]
@jrainville i noticed that dapps stopped asking for permissions therefore i cant authorise simpledapp.eth and transact there, when i try to Request STT, there is an error:

`js: Uncaught Error: invalid address
`

**Comment by jrainville** [comment]
> i noticed that dapps stopped asking for permissions therefore i cant authorise simpledapp.eth and transact there

@anastasiyaig I just tested and it happened in the commit before my branch as well. 

I'll try to fix it, but this current PR could be merged nonetheless IMO.

Let me know if you need STT to test the PR in the meantime

**Comment by anastasiyaig** [comment]
ah lets merge then :)

**Comment by jrainville** [comment]
I added the new issue here: https://github.com/status-im/status-desktop/issues/4721

**Review by alexandraB99 — APPROVED** [review]

**Pull request merged**